specs/complement.test.ts

```diff
@@ -8,6 +8,13 @@ describe('complement', () => {
     const f = complement(even)
     eq(f(8), false)
     eq(f(13), true)
+    eq(f(-1), true)
+    eq(f(-4), false)
+    eq(f(0), false)
+    eq(f(NaN), false)
```

Comment:
eq(f(NaN), false)
    eq(f(NaN), false) 
    don't add such test cases bcz 
    (i)NaN%2 == NaN 
    and NaN!== 0 //true
    Hence NaN is not even  so the even function that u have declared will return false and it's complement should 
   return true
   (ii) if we have passes the same argument for odd function then also it (the complement function) would have returned false
   so adding cases that rely on the properties of operators  and giving results that are kind of conflicting
   like in our case
   f = complement(even)
   f(NaN) // true 
   f' = complement(odd) (given that we have declared odd function as one which returns true if remainder is 1)
   f'(NaN) // true
   so in actual it is not showing the properties of complement function rather it is creating confusion
   so better we add cases like:- 
   complement(isNaN)(NaN) 
   P.S. same applies for infinity